### ISSUE TITLE
Accessing relative path not supported

### ISSUE THREAD

**Issue by Lukasz994** [microsoft/azure-pipelines-agent#2439]
## Having issue with YAML?
Please log an issue at [Azure-Pipelines-YAML](https://github.com/Microsoft/azure-pipelines-yaml). Over there we discuss YAML templates, samples for Azure Pipelines, and designs for upcoming YAML features. Also a place for the community to share best practices, ideas, and so on. File suggestions and issues here if they're specific to YAML pipelines.

## Having issue with Tasks?
Log an issue at [Azure-Pipelines-Tasks](https://github.com/Microsoft/azure-pipelines-tasks). It contains all of the in-box tasks we ship with Azure-Pipelines/VSTS/TFS. If you're having issues with tasks in Build/Release jobs (e.g. unreasonable task failure) please log an issue there.

## Having issue with software on Hosted Agent?
Log an issue at [Hosted Agent Image Repository](https://github.com/Microsoft/vsts-image-generation). It contains the VM image used in the Azure Pipelines Hosted Agent Pool. If you're having Build/Release failures that seems like they are related to software installed on the Hosted Agent (e.g. the `dotnet` SDK is missing or the Azure SDK is not on the latest version) please log an issue there.

## Having generic issue with Azure-Pipelines/VSTS/TFS?
Please report it on [Developer Community](https://developercommunity.visualstudio.com/spaces/21/index.html)

## Have you tried troubleshooting?
[Troubleshooting doc](https://www.visualstudio.com/en-us/docs/build/troubleshooting)

## Agent Version and Platform
Version of your agent? 2.144.0/2.144.1/...
Microsoft Hosted Ubuntu 16.04

OS of the machine running the agent? OSX/Windows/Linux/...
Ubuntu 16.04

## Azure DevOps Type and Version
dev.azure.com (formerly visualstudio.com) or on-premises TFS/Azure DevOps Server?
Azure DevOps Server
If on-premises, which release? 2015.0, 2017.1, 2019 RC2, etc.

If dev.azure.com, what is your organization name? https://dev.azure.com/{organization} or https://{organization}.visualstudio.com

## What's not working?
Please include error messages and screenshots.

This is quite strange issue. I am working on a project that uses JavaScript and rather complex folder structure. For sake of simplicity lets assume it looks like this:
PROJECT
│   config.js
└─── modulesDir1
│   │   module01
│   │
│   └─── tests
│             │test01.js
└─── modulesDir2
              │  module02

Building & compiling of this project works well, problem is when I get to unit tests. When build agent starts to process test01.js I have a a definition which loads module02 I have is specified as:
module = require('../../modulesDir2/module02')

Agent drop an error stating: 
Error: Cannot find module '../../modulesDir2/module02'

Am I wrong in assuming that '../../' is a normal way to specify relative path? I cannot think of any workaround that wouldn't require me to restructure my whole code base...  Any help would be appreciated.

**Comment by jtpetty** [comment]
@Lukasz994 - I apologize for the delay in responding to your issue. Are you still having this issue?

**Comment by Lukasz994** [comment]
@jtpetty Thanks for getting back to me. I have figured that actually Azure Pipeline Agent had issues with interpreting module names without ".js" extension. In the end I needed to preform non-trivial task of going through whole code base and updating all references. 

I did raise a ticket with MS in relation to this issue but havent been given definitive answer as to why pipeline agent doesnt interpret modules without .js extension (given Linux and Windows support it). If you'd be able to shed some light on this I'd much appreciate it. In any case I think we can close this issue.
